tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c

```diff
@@ -0,0 +1,470 @@
+}
+
+static int test_fp_storage_reset_perform(void)
+{
+	if (storage_reset_pass) {
+		mark_test_as_finished(RESET_TEST_FINISHED_INTERRUPTED_RESET_BEGIN_BIT_POS);
+		storage_reset_pass = false;
+		ztest_test_pass();
+	}
+
+	return 0;
+}
+
+static void test_fp_storage_reset_prepare(void) {}
+
+/* The name starts with '_0_' so that this module's struct would be first in section (linker sorts
+ * entries by name) and its callbacks would be call before other's modules callbacks.
+ */
+FP_STORAGE_MANAGER_MODULE_REGISTER(_0_test_reset, test_fp_storage_reset_perform,
+				   test_fp_storage_reset_prepare);
+
+static void test_section_placement(void)
+{
+	struct fp_storage_manager_module *module;
+
+	STRUCT_SECTION_GET(fp_storage_manager_module, 0, &module);
+
+	zassert_equal_ptr(test_fp_storage_reset_perform, module->module_reset_perform,
+			  "Invalid order of elements in section");
+	zassert_equal_ptr(test_fp_storage_reset_prepare, module->module_reset_prepare,
+			  "Invalid order of elements in section");
+
+	mark_test_as_finished(RESET_TEST_FINISHED_SECTION_PLACEMENT_BIT_POS);
+}
+
+static int settings_validate(settings_load_direct_cb cb)
+{
+	int err;
+	int err_cb = 0;
+
+	err = settings_load_subtree_direct(NULL, cb, &err_cb);
+	if (err) {
+		return err;
+	}
+
+	if (err_cb) {
+		return err_cb;
+	}
+
+	return 0;
+}
+
+static int settings_validate_empty_cb(const char *key, size_t len, settings_read_cb read_cb,
+				      void *cb_arg, void *param)
+{
+	int *err = param;
+
+	*err = -EFAULT;
+	return *err;
+}
+
+static int fp_settings_data_validate_empty_cb(const char *key, size_t len, settings_read_cb read_cb,
+					      void *cb_arg, void *param)
+{
+	int *err = param;
+
+	if (!strncmp(key, SETTINGS_RESET_TEST_STAGE_FULL_NAME,
+		     sizeof(SETTINGS_RESET_TEST_STAGE_FULL_NAME))) {
+		return 0;
+	}
+
+	if (!strncmp(key, SETTINGS_RESET_IN_PROGRESS_FULL_NAME,
+		     sizeof(SETTINGS_RESET_IN_PROGRESS_FULL_NAME))) {
+		int rc;
+		bool reset_in_progress;
+
+		if (len != sizeof(reset_in_progress)) {
+			*err = -EINVAL;
+			return *err;
+		}
+
+		rc = read_cb(cb_arg, &reset_in_progress, len);
+		if (rc < 0) {
+			*err = rc;
+			return *err;
+		}
+
+		if (rc != len) {
+			*err = -EINVAL;
+			return *err;
+		}
+
+		if (reset_in_progress) {
+			*err = -EFAULT;
+			return *err;
+		}
+
+		return 0;
+	}
+
+	*err = -EFAULT;
+	return *err;
+}
+
+static void fp_ram_clear(void)
+{
+	fp_storage_ak_ram_clear();
+	fp_storage_pn_ram_clear();
+	fp_storage_manager_ram_clear();
+}
+
+static void setup_fn(void)
+{
+	int err;
+
+	err = settings_validate(settings_validate_empty_cb);
+	zassert_ok(err, "Settings have not been fully deleted");
+
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+}
+
+static int settings_get_name_cb(const char *key, size_t len, settings_read_cb read_cb, void *cb_arg,
+				void *param)
+{
+	char *name = param;
+
+	strncpy(name, key, SETTINGS_MAX_NAME_LEN);
+	return -EALREADY;
```

Comment:
Please add a comment to inform why we actually return error here (that it's used to stop iterating over settings record)